### PR TITLE
[GAUDISW-228042] Add support for dynamic vLLM kv-cache quantization

### PR DESCRIPTION
**Pull request by dudilester**

### CODE REVIEW

File: vllm_gaudi/extension/ops.py [hunk 1]
```diff
@@ -173,15 +174,23 @@ def flat_pa_mla(query, key_cache, value_cache, block_list, block_mapping, block_
 
 def flat_pa(query, key_cache, value_cache, block_list, block_mapping, block_bias, block_groups, block_size, scale,
             matmul_qk_op, position_bias, matmul_av_op, batch2block_matmul_op, block2batch_matmul_op, keys_fetch_func,
-            values_fetch_func, **ignored_args):
+            values_fetch_func, k_scales, v_scales, **ignored_args):
```
Comment (nirda7) [inline vllm_gaudi/extension/ops.py]: shouldn't you set these new args to None as currently they are not sent from the main vLLM?

File: vllm_gaudi/attention/backends/hpu_attn.py [hunk 2]
```diff
@@ -255,7 +255,7 @@ def forward(
         latent_vec_k = latent_vec_k.view(-1, self.qk_rope_head_dim + self.kv_lora_rank)
 
         # write the latent and rope to kv cache
-        if kv_cache is not None and len(kv_cache) == 2:
+        if kv_cache is not None and len(kv_cache) >= 2:
```
[the remark this answers is not in the document]
Comment (dudilester) [reply vllm_gaudi/attention/backends/hpu_attn.py]: @skavulya currently the vllm support only per tensor scales for the kv-cache, since the data in the vllm kvcache is dynamically changing, due to continues batching feature, to achieve good scales vllm recommends running with static scales that were previously calculated on a calibration dataset. (https://docs.vllm.ai/en/latest/features/quantization/quantized_kvcache)
We are adding a feature to support dynamic quantization with per-token scales, this feature allows better accuracy as scales are dynamically configured according to current values in the K and V caches.
These scales in contrast to the static single float values (_k_scale, _v_scale) are dynamically changing as the vllm kvcache content changes, as well, as the cache grows the number of scales grow, scales are bf16 value per token (dynamically mapped to each token in the cache and removed when the token is removed from cache), that is the reason we are using similar mechanism as the actual kvcache data to hold these dynamic scales.

[the remark this answers is not in the document]
Comment (skavulya) [reply vllm_gaudi/attention/backends/hpu_attn.py]: Thanks @dudilester for the detailed explanation

File: vllm_gaudi/extension/utils.py [hunk 3]
```diff
@@ -39,20 +39,26 @@ def forward(self, x, dim=None, inv_head=None):
...
+
 class VLLMKVCache(torch.nn.Module):
 
     def __init__(self):
-        super(VLLMKVCache, self).__init__()
+        super().__init__()
         self.use_contiguous_pa = get_config().use_contiguous_pa
 
-    def forward(self, input, cache, slot_mapping):
+    def forward(self, input, cache, slot_mapping, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (dudilester) [reply vllm_gaudi/extension/utils.py]: added scales kwarg
